src/components/Breadcrumbs/index.js

```diff
@@ -1,27 +1,38 @@
 /* eslint-disable react/prop-types */
-import React from "react"
+import React, { useMemo } from "react"
 import Breadcrumbs from "react-router-dynamic-breadcrumbs"
 import { itemCls, wrapperCls } from "./styles"
 import { cx } from "emotion"
-
-const routesList = {
-  "/login": "Login",
-  "/register": "Register",
-  "/visits/details": null,
-  "/visits/details/:id": "View a single Visit",
-  "/visits/create-new": "New Visit",
-  "/visits/edit": null,
-  "/visits/edit/:id": "Edit a Visit",
-  "/visits": "All Visits",
-  "/patients": "Patients",
-  "/patients/details": null,
-  "/patients/details/:id": "Patient Details",
-  "/patients/create-new": "Create New",
-  "/patients/edit": null,
-  "/patients/edit/:id": "Patient Edit"
-}
+import { useI18n } from "providers/I18n"
 
 function BreadcrumbsComponent() {
+  const { t } = useI18n()
+
+  const routesList = useMemo(() => {
+    return {
+      "/login": `${t("int.login")}`,
+      "/register": "Register",
+      "/visits/details": null,
+      "/visits/details/:id": `${t("int.view_single_visit")}`,
+      "/visits/create-new": `${t("int.create_new_visit")}`,
+      "/visits/edit": null,
+      "/visits/edit/:id": `${t("int.edit_a_visit")}`,
+      "/visits": `${t("int.allvisits")}`,
+      "/patients": `${t("int.patients")}`,
+      "/patients/details": null,
+      "/patients/details/:id": `${t("int.patient_details")}`,
+      "/patients/create-new": `${t("int.patients_create_new")}`,
+      "/patients/edit": null,
+      "/patients/edit/:id": `${t("int.patient_edit")}`,
+      "/professionals": `${t("int.professionals")}`,
+      "/professionals/details": null,
+      "/professionals/details/:id": `${t("int.professional_details")}`,
+      "/professionals/create-new": `${t("int.professional_create_new")}`,
+      "/professionals/edit": null,
+      "/professionals/edit/:id": `${t("int.professional_edit")}`
+    }
+  }, [])
```

Comment:
t needs to exist in this useMemo dependency array. Eslint didn't warn you about it?
The reason is that if we add t in the deps array, our translations are going to be recalculated when the language changes because t is being updated when the current language is changed.
You can check it yourself. Without doing anything try and update the selected language. Then add t to the deps array and change the selected language. What do you notice in the breadcrumb in both cases?

Reply:
so this way the routesList will only be recomputed when the i18n index file or the translations file changes?

Reply:
the i18n index file and the translation files cannot be changed by the client. We can changed them during development and when the project is built and loaded to a browser, those files cannot be changed. 
The client can only change the language that they currently view the site in. Right now we have english and greek. When they change the selected language the reference of t is updated (this is taken care of in i18n/index.js). If a useMemo has t in its dependency array it will execute again. So the routesList will be recalculated with the translations of the newly selected language.